jena-base/src/main/java/org/apache/jena/atlas/io/BlockUTF8.java

```diff
@@ -135,13 +135,32 @@ private static void toCharsBuffer(ByteBuffer bb, CharBuffer cb) {
             if ( (x & 0xF8) == 0xF0 ) {
                 // Looking like 4 byte character.
                 // 11110zzz => 4 bytes.
-                int ch = readMultiBytes(bb, x & 0x08, 4);
+                int ch = readMultiBytes(bb, x & 0x07, 4);
 
                 char chars[] = Character.toChars(ch);
                 cb.put(chars);
                 idx += 4;
                 continue;
             }
+            if ( (x & 0xFC) == 0xF8 ) {
+                // 111110zz => 5 bytes.
+                int ch = readMultiBytes(bb, x & 0x03, 5);
+
+                char chars[] = Character.toChars(ch);
+                cb.put(chars);
+                idx += 5;
+                continue;
+            }
+            if ( (x & 0xFE) == 0xFC ) {
+                // Looking like 4 byte character.
+                // 11110zzz => 4 bytes.
```

Comment:
Shouldn't this one be 6 bytes? (bit late here, bit-shift-fu might be failing me :sleeping: )

Reply:
Yes, it should.